Refactor to simpler base with ability to call predefined or custom prompts

mmnavarr opened the pull request:
- Refactored class approach to more module approach using functions to initialize and using module-scoped vars to store values
- Added helper methods to set engine, and get locals of the module
- Fixed __init__.py fle to import calvin as the module itself
- Added .gitignore

Commit:
Refactored to fuunction based module that initializes and holds module heads for openai access

Commit:
Smplifed global declarations

Commit:
Refactored yuh

Commit:
Merge branch 'feature/malcolm' of https://github.com/pmohun/calvin into feature/malcolm

pmohun commented on calvin/calvin.py:
```diff
@@ -1,151 +1,146 @@
+
+
+# Initialize module
+def initialize(_api_key, _engine):
```
Let's define engine as an optional parameter here using `_engine = "davinci"` or `engine = engine`.

This will allow us to call calvin.initializat('api-key') without the need to define engine. We can define it if we'd like, but it will be optional.

pmohun commented on calvin/calvin.py:
```diff
@@ -1,151 +1,146 @@
+    return response.raise_for_status()
+
+def complete_prompt(prompt):
+  payload = {
```
How would the end-user go about adjusting the hyperparameter settings here? They would need to call `create_completion` and pass it a json, no?

If that's the case, we should set these defaults as something more user friendly. `max_tokens = 5` won't produce very useful prompts out of the box. 

I'd suggest:
max_tokens = 150
temperature = 0.9
top_p = 1
n = 1

pmohun commented on calvin/calvin.py:
```diff
@@ -1,151 +1,146 @@
+
+# Search POST
+# Perform a semantic search over a list of documents.
+def search(payload):
```
Have we tested this? 
![image](https://user-images.githubusercontent.com/11607499/88253212-2876ab00-cc7f-11ea-8ae4-a1aa0fb68331.png)

Seems really powerful.

pmohun committed:
Update calvin.py

added preset for _engine in initialize method
updated hyperparameter settings for complete_prompt method
